src/components/pages/Dashboard.tsx

```diff
@@ -25,29 +24,41 @@ export const Dashboard: React.FC = () => {
         debtToken: addresses.debtToken,
     });
 
-    const checkIfTokenAdded = useCallback(() => {
-        if (!addresses.debtToken) return false;
-        return Boolean(localStorage.getItem(`token_${addresses.debtToken}`));
-    }, [addresses.debtToken]);
+    useEffect(() => {
+        if (!isConnected || !address || !addresses.debtToken) return;
 
-    const dismissTokenBanner = () => {
-        if (addresses.debtToken) {
-            localStorage.setItem(`token_${addresses.debtToken}`, 'true');
+        if (
+            !addresses.debtToken.startsWith('0x') ||
+            addresses.debtToken.length !== 42
```

Comment:
You can use `isAddress` to check if it is the wallet address.
https://viem.sh/docs/utilities/isAddress